woocommerce/woocommerce
# Introduce a remote inbox rule for Woo Express sites

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Introduces a new rule to allow remote Inbox notifications from WooCommerce.com to conditionally reach (or not reach) sites on Woo Express plans.

The rule supports an optional `plan` parameter that makes it possible to target a specific Woo Express plan. 



### How to test the changes in this Pull Request:

1. Add `define('WC_ADMIN_DEBUG_RULE_EVALUATOR', true);` to your `wp-config.php` file. This enables RIN logging under **WooCommerce > Status > Logs**.
2. Create a file called `rin.json` in your root directory (same directory as the `wp-config.php` file. Example contents below.
3. Temporarily modify the `DATA_SOURCES` constant in `DataSourcePoller.php` to point to your local json file. - [See Github](https://github.com/woocommerce/woocommerce/blob/670e2c06693e5766e2be5157f9b2644b10367658/plugins/woocommerce/src/Admin/RemoteInboxNotifications/DataSourcePoller.php#L17-L19)
4. Force a refresh of the remote messages from your json file. `wp cli event run wc_admin_daily`. Remember to update the `slug` in your json when you want to test a new scenario.
5. You can emulate plan detection locally using [these functions](https://github.com/Automattic/wc-calypso-bridge/blob/master/class-wc-calypso-bridge-dotcom-features.php).

### Scenarios to test:

- Target a Woo Express site without a specified plan. 
  - Confirm that a message is added to the Inbox when `value` is true. 
  - Confirm that a message is not added when the it is false.
- Target a Woo Express plan that is valid using the `plan` parameter, e.g. `trial` or `performance`. 
  - Confirm that a message is added to the Inbox when `value` is true. 
  - Confirm that a message is not added when the it is false.
- Target a Woo Express plan name that's invalid e.g. `invalid`. 
  - Confirm that a message is not added when `value` is true (because plan name detection fails).
  - When `value` is false, remember that the intended behavior is that the message should be added to all plans other than the specified one, **provided that we're still on a Woo Express plan**. 

Example JSON file:

```
[
   {
      "slug":"woox-test-1",
      "type":"info",
      "status":"unactioned",
      "is_snoozable":0,
      "source":"woocommerce.com",
      "locales":[
         {
            "locale":"en_US",
            "title":"This is just a test.",
            "content":"Just some random content."
         }
      ],
      "actions":[
         {
            "name":"some-action",
            "locales":[
               {
                  "locale":"en_US",
                  "label":"Share feedback"
               }
            ],
            "url":"https:\/\/automattic.com",
            "url_is_admin_query":false,
            "is_primary":true,
            "status":"actioned"
         }
      ],
      "rules":[
         {
            "type":"is_woo_express",
            "plan":"trial",
            "value":true
         }
      ]
   }
]
```


### Changelog entry

Introduced a new Inbox rule to allow Inbox messages to be added remotely to WordPress.com stores on the Woo Express plan.





<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
